### PR TITLE
[Wallet] Redesign - Add header components

### PR DESCRIPTION
**Pull request by i1skn**
### Description
* add `TopBarIconButton` and `TopBarTextButton`.
* add generic `HeaderTitleWithSubtitle` 
* refresh header with balance

### Tested

Android emulator

### Backwards compatibility

Yes

#### 1
Screenshot_1589378972
#### 2
Screenshot_1589379000
#### 3
Screenshot_1589378963
#### 4
Screenshot_1589378978
#### 5 (when top bar button is pressed)
Screenshot_1589459370

### CODE REVIEW

File: packages/mobile/src/components/BackButton.v2.tsx
Comment (aaronmgdr) [inline packages/mobile/src/components/BackButton.v2.tsx]: why not just pass navigateBack directly to BackButton?

File: packages/mobile/src/navigator/TopBarButton.v2.tsx
Comment (aaronmgdr) [inline packages/mobile/src/navigator/TopBarButton.v2.tsx]: why this change? 

why remove Touchable? it handles having android phones have the nice ripple.

File: packages/react-components/components/BackButton.v2.tsx
Comment (aaronmgdr) [inline packages/react-components/components/BackButton.v2.tsx]: im not sure about using these variables in new design. since everything is multiples of 8. the +7 worries me

File: packages/mobile/src/navigator/TopBarButton.v2.test.tsx
Comment (aaronmgdr) [inline packages/mobile/src/navigator/TopBarButton.v2.test.tsx]: hmm I think we should avoid changing interfaces

File: packages/mobile/src/navigator/TopBarButton.v2.tsx
Comment (aaronmgdr) [inline packages/mobile/src/navigator/TopBarButton.v2.tsx:28]: hmm the buttons i reference in the header when i made this were def `500`

Comment (i1skn) [reply packages/mobile/src/navigator/TopBarButton.v2.tsx:28]: @coreycelo did some changes in the design, so I've reflected them here.

Comment (aaronmgdr) [reply packages/mobile/src/navigator/TopBarButton.v2.tsx:28]: ok cool

File: packages/react-components/icons/BackChevron.v2.tsx
Comment (jmrossy) [inline packages/react-components/icons/BackChevron.v2.tsx]: Setting default static props here like we do for some other icons may be a nice addition

File: packages/react-components/components/BackButton.v2.tsx
Comment (jmrossy) [reply packages/react-components/components/BackButton.v2.tsx]: +1, setting up new variables may be nice.
But also, I find variables for things like padding to be unhelpful. How often will we say 'reduce padding across the app by 2'. Even if we did, unlikely that most things would line up right after the var change

File: packages/react-components/components/BackButton.v2.tsx
Comment (jmrossy) [inline packages/react-components/components/BackButton.v2.tsx]: Maybe not a problem but just a note, I think hitslop only works if the parent container is sufficiently large (ie small parent will cut off hitslop area)

File: packages/mobile/src/navigator/TopBarButton.v2.test.tsx
Comment (jmrossy) [reply packages/mobile/src/navigator/TopBarButton.v2.test.tsx]: I think it should be case-by-case. A title prop does make more sense to me than a child component

File: packages/mobile/src/navigator/Headers.v2.tsx
Comment (jmrossy) [inline packages/mobile/src/navigator/Headers.v2.tsx:93]: I find the possibility of null balance creates ugly conditionals all across the app and I wonder if we'd be better off just defaulting to 0. We'd lose a signal that says 'I don't know the balance' but it would simplify many places

**Review by jmrossy — COMMENTED** [review]
Mostly looks great!

File: packages/mobile/src/navigator/TopBarButton.v2.test.tsx
Comment (i1skn) [reply packages/mobile/src/navigator/TopBarButton.v2.test.tsx:8]: I agree here with Rossy, plus it was introduced only in `v2`, so potentially will only affect our WIP branches. @aaronmgdr WDYT?

File: packages/mobile/src/navigator/Headers.v2.tsx
Comment (i1skn) [reply packages/mobile/src/navigator/Headers.v2.tsx:93]: I think, this is a bit out of the scope of this PR. But I agree, we should come up with a convenient approach how we represent unknown balance.

**Comment by i1skn** [comment]
@aaronmgdr I've reworked `TopBarButton` and split it into `TopBarIconButton` and `TopBarTextButton`. I know that you have an opened PR, which relies on it and I'm happy to wait until we merge your one first.

File: packages/react-components/components/BackButton.v2.tsx
Comment (i1skn) [reply packages/react-components/components/BackButton.v2.tsx]: I've moved this logic to `mobile` package. Regarding the padding itself I've changed it to `+6` (correlates with the design) to the standard padding we have.

Screenshot 2020-05-14 at 12 56 26

**Review by aaronmgdr — APPROVED** [review]

**Pull request merged**
